### PR TITLE
Funnel -> core

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/funnel/Funnel.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	public Rotation2d getPosition() {
+		return funnelStuff.positionSignal().getLatestValue();
+	}
+
+	public void setTargetPosition(double rotations) {
+		this.targetPosition = Rotation2d.fromRotations(getPosition().getRotations() + rotations);
+	}
+
+	public boolean isAtPosition(Rotation2d position) {
+		return (getPosition().getRotations() - position.getRotations() <= 5);
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/subsystems/funnel/Funnel.java]: no paranthesseseses

File: src/main/java/frc/robot/subsystems/funnel/FunnelStuff.java
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/funnel/FunnelStuff.java:1]: IRequest?

File: src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	private final static SparkLimitSwitch.Type FORWARD_LIMIT_SWITCH_TYPE = SparkLimitSwitch.Type.kNormallyOpen;
+
+
+	public static FunnelStuff generateFunnelStuff(String logPath) {
+		SparkMaxWrapper wrapper = new SparkMaxWrapper(IDs.CANSparkMAXIDs.FUNNEL);
+		SysIdRoutine.Config config = new SysIdRoutine.Config();
+		BrushlessSparkMAXMotor motor = new BrushlessSparkMAXMotor(logPath, wrapper, config);
+
+		Supplier<Double> voltage = () -> wrapper.getBusVoltage() * wrapper.getAppliedOutput();
+		SparkMaxDoubleSignal voltageSignal = new SparkMaxDoubleSignal("voltage", voltage);
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java:33]: one line

**Review by hillelv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/funnel/FunnelStuff.java
Comment (drorta) [reply src/main/java/frc/robot/subsystems/funnel/FunnelStuff.java:1]: why?

File: src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java (hunk 2 above)
Comment (drorta) [reply src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java:33]: again, why?

File: src/main/java/frc/robot/subsystems/funnel/Funnel.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+		commandsBuilder = new FunnelCommandsBuilder(this);
+		update();
+	}
+
+	public FunnelCommandsBuilder getCommandsBuilder(){
+		return commandsBuilder;
+	}
+
+	public boolean isObjectInShooter() {
```
Comment (greenblitz4590) [inline src/main/java/frc/robot/subsystems/funnel/Funnel.java]: isNoteinShooter?

File: src/main/java/frc/robot/subsystems/funnel/Funnel.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		return shooterDigitalInputInputs.debouncedValue;
+	}
+
+	public boolean isObjectInAmp() {
+		return ampDigitalInputInputs.debouncedValue;
+	}
+
+	public void update() {
```
Comment (greenblitz4590) [inline src/main/java/frc/robot/subsystems/funnel/Funnel.java]: rename to updateSignls or update inputs

File: src/main/java/frc/robot/subsystems/funnel/Funnel.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	public void setBrake(boolean brake) {
+		motor.setBrake(brake);
+	}
+
+	@Override
+	protected void subsystemPeriodic() {
+		update();
+		Logger.processInputs(funnelStuff.shooterDigitalInputLogPath(), shooterDigitalInputInputs);
+		Logger.processInputs(funnelStuff.ampDigitalInputLogPath(), ampDigitalInputInputs);
+		Logger.recordOutput(funnelStuff.logPath() + "IsObjectInShooter", isObjectInShooter());
```
Comment (greenblitz4590) [inline src/main/java/frc/robot/subsystems/funnel/Funnel.java]: same here

**Review by greenblitz4590 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java (hunk 2 above)
Comment (hillelv) [reply src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java:33]: we will merge a function for this in the template

Comment (drorta) [reply src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java:33]: wdym?

File: src/main/java/frc/robot/constants/IDs.java [hunk 6]
```diff
@@ -10,4 +12,8 @@ public class IDs {
 		PowerDistribution.ModuleType.kRev
 	);
 
+	public static class CANSparkMAXIDs {
+		public static final SparkMaxDeviceID FUNNEL = new SparkMaxDeviceID(1, CANSparkLowLevel.MotorType.kBrushless);
```
Comment (YHerm) [inline src/main/java/frc/robot/constants/IDs.java:17]: sapce above and below

File: src/main/java/frc/robot/subsystems/funnel/FunnelCommandsBuilder.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                () -> funnel.setPower(power.getAsDouble()),
+                interrupted -> funnel.stop(),
+                () -> false,
+                funnel
+        ).withName("Set funnel power by supplier");
+    }
+    //@formatter:on
+
+	public Command stopFunnel() {
+		return new InstantCommand(funnel::stop, funnel).withName("Stop funnel");
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/funnel/FunnelCommandsBuilder.java]: run command

Comment (drorta) [reply src/main/java/frc/robot/subsystems/funnel/FunnelCommandsBuilder.java]: wdym?

File: src/main/java/frc/robot/subsystems/funnel/Funnel.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+public class Funnel extends GBSubsystem {
+
+	private final IMotor motor;
+	private final FunnelStuff funnelStuff;
+	private final FunnelCommandsBuilder commandsBuilder;
+	private final IDigitalInput shooterDigitalInput;
+	private final IDigitalInput ampDigitalInput;
+	private final DigitalInputInputsAutoLogged shooterDigitalInputInputs;
+	private final DigitalInputInputsAutoLogged ampDigitalInputInputs;
```
Comment (Nitay4) [inline src/main/java/frc/robot/subsystems/funnel/Funnel.java:18]: maby put them in stuff

**Review by Nitay4 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/funnel/Funnel.java (hunk 8 above)
Comment (drorta) [reply src/main/java/frc/robot/subsystems/funnel/Funnel.java:18]: already in stuff, and easier this way

**Review by YHerm — COMMENTED** [review]
u have conflicts

**Pull request closed without merging**